src/trustline_asset.js

```diff
@@ -0,0 +1,236 @@
+import clone from 'lodash/clone';
+import padEnd from 'lodash/padEnd';
+import trimEnd from 'lodash/trimEnd';
+import xdr from './generated/stellar-xdr_generated';
+import { Keypair } from './keypair';
+import { StrKey } from './strkey';
+
+/**
+ * TrustLineAsset class represents the asset referenced by a trustline to either
+ * a liquidity pool, the native asset (`XLM`) or an issued asset with an asset
+ * code / issuer account ID pair.
+ *
+ * In case of the native asset, the code will represent `XLM` while the issuer
+ * and liquidityPoolId will be empty. For an issued asset, the code and issuer
+ * will be valid and liquidityPoolId will be empty. For liquidity pools, the
+ * liquidityPoolId will be valid and the remaining two fields will be empty.
+ *
+ * @constructor
+ * @param {string} code - The asset code.
+ * @param {string} issuer - The account ID of the asset issuer.
+ * @param {string} liquidityPoolId - The ID of the liquidity pool in string 'hex'.
+ */
+export class TrustLineAsset {
+  constructor(code, issuer, liquidityPoolId) {
+    if (!code && !issuer && !liquidityPoolId) {
+      throw new Error('Must provide either code, issuer or liquidityPoolId');
+    }
+
+    if ((code || issuer) && liquidityPoolId) {
+      throw new Error(
+        'Must provide either code (and optionally issuer) or liquidityPoolId but not both'
+      );
+    }
+
+    // Validate code & issuer.
+    if ((code || issuer) && !/^[a-zA-Z0-9]{1,12}$/.test(code)) {
+      throw new Error(
+        'Asset code is invalid (maximum alphanumeric, 12 characters at max)'
+      );
+    }
+    if (code && String(code).toLowerCase() !== 'xlm' && !issuer) {
+      throw new Error('Issuer cannot be null');
+    }
+    if (issuer && !StrKey.isValidEd25519PublicKey(issuer)) {
+      throw new Error('Issuer is invalid');
+    }
+
+    // Test if liquiditry pool ID is a hash.
+    if (liquidityPoolId && !/^[a-fA-F0-9]{64}$/.test(liquidityPoolId)) {
```

Comment:
Can you confirm that there won't be casing issues here? The XDR always uses lowercase letters in its hashes and I just want to make sure that doesn't cause problems. To be extra safe, we can do 
```js
this.liquidityPoolId = liquidityPoolId.toLowerCase();
```
below, instead.

Reply:
Added this new validation in https://github.com/stellar/js-stellar-base/pull/454/commits/f32e27277343e6cc4240c95ac241878ce9e50781.